libraries/CurieBLE/src/BLEProfile.cpp

```diff
@@ -0,0 +1,513 @@
+#include <errno.h>
+#include "BLEProfile.h"
+#include "BLEPeripheral.h"
+#include "BLECentralRole.h"
+#include "BLEPeripheralRole.h"
+
+// Only for peripheral
+ssize_t profile_read_process(struct bt_conn *conn,
+                             const struct bt_gatt_attr *attr,
+                             void *buf, uint16_t len,
+                             uint16_t offset)
+{
+    const unsigned char *pvalue;
+    BLEAttribute *bleattr = (BLEAttribute *)attr->user_data;
+    BLECharacteristic* blecharacteritic;
+    BLEAttributeType type = bleattr->type();
+    if (BLETypeCharacteristic != type)
+    {
+        return 0;
+    }
+    blecharacteritic = (BLECharacteristic*)bleattr;
+    pvalue = blecharacteritic->value();
+    return bt_gatt_attr_read(conn, attr, buf, len, offset, pvalue,
+                             blecharacteritic->valueLength());
+}
+
+// Only for peripheral
+ssize_t profile_write_process(struct bt_conn *conn,
+                                     const struct bt_gatt_attr *attr,
+                                     const void *buf, uint16_t len,
+                                     uint16_t offset)
+{
+    pr_info(LOG_MODULE_BLE, "%s1", __FUNCTION__);
+    BLEAttribute *bleattr = (BLEAttribute *)attr->user_data;
+    BLECharacteristic* blecharacteritic;
+    BLEAttributeType type = bleattr->type();
+    BLECentralHelper central = BLEPeripheralRole::instance()->central();
+    if ((BLETypeCharacteristic != type) || 0 != offset)
+    {
+        return 0;
+    }
+    
+    blecharacteritic = (BLECharacteristic*)bleattr;
+    blecharacteritic->setValue(*((BLEHelper *)&central), (const uint8_t *) buf, len);
+    
+    return len;
+}
+
+
+// Only for central
+uint8_t profile_notify_process (struct bt_conn *conn,
+                             struct bt_gatt_subscribe_params *params,
+                             const void *data, uint16_t length)
+{
+    BLEPeripheralHelper* peripheral = BLECentralRole::instance()->peripheral(conn);// Find peripheral by bt_conn
+    BLEAttribute* notifyatt = peripheral->attribute(params); // Find attribute by params
+    BLECharacteristic *chrc = (BLECharacteristic *)notifyatt;
+    
+    //assert(notifyatt->type() == BLETypeCharacteristic);
```

Comment:
For the project didn't define the assert. Will uncomment it after we define the interface. And this is in my plan. Try to let function works first.